File: src/main/java/org/tindalos/principle/infrastructure/service/jdepend/JDependPackageAnalyzer.java
package org.tindalos.principle.infrastructure.service.jdepend;

import java.util.Collection;
import java.util.List;

import jdepend.framework.JavaPackage;

import org.tindalos.principle.domain.checker.PackageAnalyzer;
import org.tindalos.principle.domain.core.Package;
import org.tindalos.principle.domain.core.DesignQualityCheckConfiguration;

public class JDependPackageAnalyzer implements PackageAnalyzer {
    
    private final JDependRunner jDependRunner;
    private final PackageListFactory packageListFactory;

    public JDependPackageAnalyzer(JDependRunner jDependRunner, PackageListFactory packageListFactory) {
        this.jDependRunner = jDependRunner;
        this.packageListFactory = packageListFactory;
    }

    public List<Package> analyze(DesignQualityCheckConfiguration parameters) {
        Collection<JavaPackage> analyzedPackages = jDependRunner.getAnalyzedPackages(parameters.getBasePackage());
        return packageListFactory.build(analyzedPackages);
    }

}

File: src/main/scala/org/tindalos/principle/infrastructure/service/jdepend/JDependPackageAnalyzer.scala
package org.tindalos.principle.infrastructure.service.jdepend

import jdepend.framework.JavaPackage
import org.tindalos.principle.domain.core.{AnalysisPlan, Package}

import scala.collection.immutable.Seq

object JDependPackageAnalyzer {

  def buildAnalyzerFn(produceJavaPackageList: (String, Boolean) => Seq[JavaPackage]
                    , transform: (String, Seq[JavaPackage]) => Seq[Package]) =

    (rootPackage: String) => {
      val analyzedPackages = produceJavaPackageList(rootPackage,true)
      transform(rootPackage, analyzedPackages)
    }

}